@here/harp-materials/lib/SolidLineMaterial.ts

```diff
@@ -153,6 +175,15 @@ uniform vec2 drawRange;
 uniform float dashSize;
 uniform float gapSize;
 uniform vec3 dashColor;
+
+#define DASHES_SQUARE ${LineDashesModes.DASHES_SQUARE}
+#define DASHES_ROUND ${LineDashesModes.DASHES_ROUND}
+#define DASHES_DIAMOND ${LineDashesModes.DASHES_DIAMOND}
+
+#ifndef DASHES_MODE
+#define DASHES_MODE ${LineDashesModes.DASHES_SQUARE}
```

Comment:
I would rather prefer to setup (default) define via shader params, instead of injecting it into the shader code, this way it will be clear from TypeScript side which DASH_MODE is set by default.
Imagine somebody calling a getter (`get dashMode()`), without setting it in params, we have fallback there, but the ralation with shader code it is not that clear, if default will change we may get into regression. If we set it here:
```
       // Setup default defines.
        const defines: { [key: string]: any } = {
            CAPS_MODE: LineCapsModes.CAPS_ROUND,
            DASHES_MODE: LineDashesMode.DASHES_SQUARE
        }
```
then situation is clear, no need for `ifndef DASHES_MODE .... ` in shader, no need for special handling in getter.

Reply:
The idea was to not have the overhead of setting this #define for every line, not just the dashed ones. Will add the dashes to the defaults.